### PR TITLE
Implemented Rollbar Items API to fetch exceptions weekly.

### PR DESCRIPTION
**Pull request by KaiwalyaPataskar**

### CODE REVIEW

File: lib/tasks/fetch_rollbar_statistics.rake [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  Repository.each do |repo|
+    load_issues(page = 1, repo) if repo.rollbar_access_token.present?
+  end
+end
+
+def load_issues(page = 1, repo)
+  url = "https://api.rollbar.com/api/1/items?page=#{page}&environment=production"
+  headers = { 'X-Rollbar-Access-Token' => repo.rollbar_access_token }
+  begin
+    response = HTTParty.get(url, headers: headers, timeout: 20)
```
[the remark this answers is not in the document]
Comment (KaiwalyaPataskar) [reply lib/tasks/fetch_rollbar_statistics.rake:12]: Let's have a service for HTTParty :+1:

File: app/models/rollbar_statistic.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class RollbarStatistic
+  include Mongoid::Document
+  include Mongoid::Timestamps
+  belongs_to :repository
+  field :date
+  field :total_issues
+  field :active_issue_count
```
Comment (swapnilchincholkar) [inline app/models/rollbar_statistic.rb:7]: initialize count fields to 0 please

File: lib/tasks/fetch_rollbar_statistics.rake [hunk 3]
```diff
@@ -0,0 +1,47 @@
+desc 'Fetches & loads weekly Rollbar Items for each repo'
+task fetch_rollbar_statistics: :environment do
+  Repository.each do |repo|
+    load_issues(page = 1, repo) if repo.rollbar_access_token.present?
```
Comment (swapnilchincholkar) [inline lib/tasks/fetch_rollbar_statistics.rake:4]: @anujaware @KaiwalyaPataskar 
From the rake task, we should invoke a service, that will do all these API calls and updates per response. (probably once a week or as per schedule). 

If we have it in service and if this fetch call fails, one can re-invoke same from UI also (when we add that option).

Comment (KaiwalyaPataskar) [reply lib/tasks/fetch_rollbar_statistics.rake:4]: @swapnilchincholkar Creating a service for this isn't a big deal, but I couldn't see the use case where we would need to call **weekly** issue fetching task to run any random time & that is also from UI or somewhere else.
This task should get executed in equal intervals in order to see the appropriate comparison.

File: app/models/rollbar_statistic.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class RollbarStatistic
+  include Mongoid::Document
+  include Mongoid::Timestamps
+  belongs_to :repository
+  field :date
+  field :total_issues, type: Integer, default: 0
```
Comment (anujaware) [inline app/models/rollbar_statistic.rb:6]: @KaiwalyaPataskar Let's not assign default value to 0. We won't be able to find if rollbar return 0 or we missed it.

**Comment by anujaware** [comment]
@KaiwalyaPataskar Is it done? Please assign the PR to me once you are done.
@swapnilchincholkar Please review the PR. We can merge and release it on production if everything is ok.

**Comment by anujaware** [comment]
@swapnilchincholkar You can review this PR. Please check and let me know if we can plan to release it in this week.

File: app/models/rollbar_statistic.rb (hunk 4 above)
Comment (anujaware) [reply app/models/rollbar_statistic.rb:6]: @KaiwalyaPataskar Is this removed? I can still see the 0.

**Pull request merged**